### PR TITLE
feat(core): tags inline editor

### PR DESCRIPTION
**Pull request by pengx17**
tags inline editor and some refactor

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/T2klNLEk0wxLh4NRDzhk/439da1e3-30a9-462a-b7b4-c8e7c3b5ef17.mp4">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/T2klNLEk0wxLh4NRDzhk/439da1e3-30a9-462a-b7b4-c8e7c3b5ef17.mp4">Kapture 2024-01-31 at 23.29.11.mp4</video>

fix AFF-467
fix AFF-468
fix AFF-472
fix AFF-466

### CODE REVIEW

**Comment by pengx17** [comment]
Current dependencies on/for this PR:
* **#5809** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5809?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5769** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5769?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5748** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5748?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5739** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5739?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5729** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5729?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5804** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5804?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5803** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5803?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5748?utm_source=stack-comment).

File: packages/frontend/core/src/components/affine/page-properties/table.tsx [hunk 1]
```diff
@@ -878,7 +922,15 @@ export const PagePropertiesAddProperty = () => {
...
   );
   return (
-    <Menu rootOptions={{ onOpenChange: () => setAdding(true) }} items={items}>
+    <Menu
+      contentOptions={{
+        onClick(e) {
+          e.stopPropagation();
+        },
+      }}
+      rootOptions={{ onOpenChange: () => setAdding(true) }}
```
Comment (Brooooooklyn) [inline packages/frontend/core/src/components/affine/page-properties/table.tsx]: how about putting these two props into `useMemo`

File: packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  return (
+    <Menu
+      rootOptions={{
+        modal: true,
+      }}
+      contentOptions={{
+        onClick(e) {
+          e.stopPropagation();
+        },
+      }}
```
Comment (Brooooooklyn) [inline packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx]: `useMemo`

File: packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      }
+
+      const newTag = {
+        id: nanoid(),
+        value: name.trim(),
+        color: nextColor,
+      };
+      rotateNextColor();
+      onOptionsChange?.([...options, newTag]);
+      setTimeout(() => {
```
Comment (Brooooooklyn) [inline packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx]: Why do we need a `setTimeout` here? Maybe worth to add a comment to explain

File: packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+        side: 'bottom',
+        align: 'start',
+        sideOffset: 0,
+        avoidCollisions: false,
+        className: styles.tagsMenu,
+        onClick(e) {
+          e.stopPropagation();
+        },
+      }}
+      items={
```
Comment (Brooooooklyn) [inline packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx:353]: `useMemo`

File: packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx (hunk 3 above)
Comment (pengx17) [reply packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx]: It looks like this is not required. Removed

File: packages/frontend/component/src/ui/input/style.css.ts [hunk 5]
```diff
@@ -8,7 +8,7 @@ export const inputWrapper = style({
   color: cssVar('textPrimaryColor'),
   border: '1px solid',
   backgroundColor: cssVar('white'),
-  borderRadius: 8,
+  borderRadius: 4,
```
Comment (CatsJuice) [inline packages/frontend/component/src/ui/input/style.css.ts]: Won't this affect other components that used?

**Review by Brooooooklyn — APPROVED** [review]

**Review by CatsJuice — APPROVED** [review]

File: packages/frontend/component/src/ui/input/style.css.ts (hunk 5 above)
Comment (pengx17) [reply packages/frontend/component/src/ui/input/style.css.ts]: Forgot to change it back. lol

**Pull request merged**